### PR TITLE
Fix issue where incomplete string is encoded for alphabets other than default

### PR DESCRIPTION
**Pull request by marmegh**
Ensure we generate at least 6 characters by default and allow customization as appropriate when using the `CustomAlphabetEncoder`.

### CODE REVIEW

File: src/Microsoft.Security.Utilities/CustomAlphabetEncoder.cs [hunk 1]
```diff
@@ -71,6 +73,11 @@ public string Encode(uint data)
                 data /= (uint)alphabet.Length;
             }
 
+            while (s_sb.Length < minLength)
+            {
+                s_sb.Append(alphabet[0]);
```
Comment (cfaucon) [inline src/Microsoft.Security.Utilities/CustomAlphabetEncoder.cs:78]: Is it possible for alphabet to be length 0 here which would throw? Does the condition need to guard against that possibility? Or is that not possible whatsoever

File: src/Microsoft.Security.Utilities/CustomAlphabetEncoder.cs [hunk 2]
```diff
@@ -60,7 +62,7 @@ public CustomAlphabetEncoder(string customAlphabet = DefaultBase62Alphabet)
         /// <returns>
         /// The unsigned integer encoded using the character set with which this class was instantiated.
         /// </returns>
-        public string Encode(uint data)
+        public string Encode(uint data, int minLength = 6)
```
Comment (cfaucon) [inline src/Microsoft.Security.Utilities/CustomAlphabetEncoder.cs:65]: Does the summary or some commenting need to be present to describe why 6 is a sensible default?

File: src/Tests.Microsoft.Security.Utilities/CustomAlphabetEncoderTests.cs [hunk 3]
```diff
@@ -48,14 +42,26 @@ public void CustomAlphabetEncoder_EncodeUintWithTestCases()
...
-                    ExpectedEncoded = ""
+                    Input = (uint)123456,
+                    ExpectedEncoded = "cdfbdca"
                 },
                 new
                 {
-                    Alphabet = "abcdef",
-                    Input = (uint)123456,
-                    ExpectedEncoded = "cdfbdca"
+                    Alphabet = "abcdefghijklmnopqrstuvwxyz" +
```
Comment (cfaucon) [inline src/Tests.Microsoft.Security.Utilities/CustomAlphabetEncoderTests.cs:50]: Does these new cases capture regression cases for the bug identified that prompted this change?

File: src/Tests.Microsoft.Security.Utilities/CustomAlphabetEncoderTests.cs [hunk 4]
```diff
@@ -48,14 +42,26 @@ public void CustomAlphabetEncoder_EncodeUintWithTestCases()
...
+                               "ABCDEFGHIJKLMNOPQRSTUVWXYZ" +
+                               "1234567890" +
+                               ".-_~",
+                    Input = (uint)956747778,
+                    ExpectedEncoded = "aYB6eE"
+                },
+                new
+                {
+                    Alphabet = "abcdefghijklmnopqrstuvwxyz" +
```
Comment (cfaucon) [inline src/Tests.Microsoft.Security.Utilities/CustomAlphabetEncoderTests.cs:59]: If this is the same alphabet to be reused in multiple cases, make it a const or static readonly for reuse.

File: src/Microsoft.Security.Utilities/CustomAlphabetEncoder.cs (hunk 1 above)
Comment (marmegh) [reply src/Microsoft.Security.Utilities/CustomAlphabetEncoder.cs:78]: No, if a length 0 alphabet is passed, we switch to the default.

File: src/Tests.Microsoft.Security.Utilities/CustomAlphabetEncoderTests.cs (hunk 3 above)
Comment (marmegh) [reply src/Tests.Microsoft.Security.Utilities/CustomAlphabetEncoderTests.cs:50]: Yes, these are the two examples Eddy produced that match the reported bug. The bug occurs when using the alphabet present in spam legacy.

**Review by cfaucon — APPROVED** [review]

**Pull request merged**